plugins/inputs/mesos/mesos.go

```diff
@@ -593,11 +672,119 @@ func (m *Mesos) gatherMainMetrics(u *url.URL, role Role, acc telegraf.Accumulato
 		}
 	}
 
+	var includeTaggedFields bool
+	for _, col := range m.MasterCols {
+		if col == "framework_offers" || col == "allocator" {
+			includeTaggedFields = true
+			break
+		}
+	}
+
+	taggedFields := map[string][]TaggedField{}
+	extraTags := map[string]map[string]string{}
```

Comment:
I think we should introduce type for tags. Later it will be really hard to distinguish if wha map of string is. http://wiki.c2.com/?StringlyTyped